### PR TITLE
UCS/STATS: missing lock

### PR DESCRIPTION
**Pull request by eladpe**
add lock for filter node add

### CODE REVIEW

File: src/ucs/stats/stats.c [hunk 1]
```diff
@@ -296,6 +280,27 @@ static void ucs_stats_add_to_filter(ucs_stats_node_t *node,
...
+    if (parent == NULL) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    /* Append node to existing tree */
+    pthread_mutex_lock(&ucs_stats_context.lock);
+    ucs_list_add_tail(&parent->children[UCS_STATS_ACTIVE_CHILDREN], &node->list);
+    node->parent = parent;
+
+    status = ucs_stats_add_to_filter(node, &ucs_stats_context.root_filter_node);
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: if this one fails, need to perform cleanup

File: src/ucs/stats/stats.c [hunk 2]
```diff
@@ -296,6 +280,27 @@ static void ucs_stats_add_to_filter(ucs_stats_node_t *node,
...
         }
     }
+
+    return UCS_OK;
+}
+
+static int ucs_stats_node_add(ucs_stats_node_t *node, ucs_stats_node_t *parent)
+{
+    ucs_status_t status;
+    ucs_assert(node != &ucs_stats_context.root_node);
```
Comment (yosefe) [inline src/ucs/stats/stats.c]: pls add space line

File: src/ucs/stats/stats.c (hunk 1 above)
Comment (eladpe) [inline src/ucs/stats/stats.c]: ucs_stats_add_to_filter may fail only on malloc failure. So there is nothing to clean. when returning to the caller, node is being free.

Comment (yosefe) [inline src/ucs/stats/stats.c]: what about removing from &parent->children[UCS_STATS_ACTIVE_CHILDREN] list?

Comment (yosefe) [inline src/ucs/stats/stats.c]: also - no need to pass &ucs_stats_context.root_filter_node as parameter, the function does not use filter_tree?

Comment (eladpe) [inline src/ucs/stats/stats.c]: I didn't use it, because i used ucs_stats_context.root_filter_node directly

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3544/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3545/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1624/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1625/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1627/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3547/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1630/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3550/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by shamisp** [comment]
@yosefe Yossi.........

**Comment by yosefe** [comment]
@eladpe pls PR this for v1.2

**Comment by shamisp** [comment]
@yosefe i think we should not merge until we have corresponding v1.2 PR ready

**Comment by eladpe** [comment]
https://github.com/openucx/ucx/pull/1511

**Comment by shamisp** [comment]
awesome . THanks

**Pull request merged**